records_mover/session.py

```diff
@@ -53,7 +59,25 @@ def _infer_scratch_s3_url(session_type: str) -> Optional[str]:
     if 'aws' in cfg:
         s3_scratch_url: Optional[str] = cfg['aws'].get('s3_scratch_url')
         if s3_scratch_url is not None:
-            return s3_scratch_url
+            #
+            # Automatically add IAM username after specified directory URL
+            #
+            root_home_directories = cfg['aws'].get('root_home_directories', False)
+            if root_home_directories:
+                #
+                # TODO: Which creds should be used here?  Should this
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker